src/net.md

```diff
@@ -876,6 +877,61 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-extract-mediawiki-links]: #ex-extract-mediawiki-links
+<a name="ex-extract-mediawiki-links"/>
+## Extract all unique links from a MediaWiki markup
+
+[![reqwest-badge]][reqwest] [![regex-badge]][regex] [![cat-net-badge]][cat-net]
+
+Pull the source of a MediaWiki page using [`reqwest::get`] and then
+look for all entries of internal and external links with
+[`Regex::captures_iter`].
+
+MediaWiki syntax is described [here][MediaWiki link syntax].
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate reqwest;
+extern crate regex;
+
+use std::collections::HashSet;
+use std::io::Read;
+use regex::Regex;
+
+# error_chain! {
+#     foreign_links {
+#         Io(std::io::Error);
+#         Reqwest(reqwest::Error);
+#         Regex(regex::Error);
+#     }
+# }
+# 
+fn run() -> Result<()> {
+    let mut content = String::new();
+    reqwest::get(
+        "https://en.wikipedia.org/w/index.php?title=Rust_(programming_language)&action=raw",
+    )?
+        .read_to_string(&mut content)?;
+
+    let mut links = Regex::new(r#"\[\[(?P<link>[^\[\]|]*)[^\[\]]*\]\]"#)?
```

Comment:
i would suggest to have a single regex concatenated with | 
to make the regex readable please use the multiline (whitespace insensitive) regex definition like in https://brson.github.io/rust-cookbook/basics.html#run-an-external-command-and-process-stdout (you can split the regex into smaller parts each with comment starting with #)

having different capture groups would actually be beneficial as you will be able to match on them separately and implement the case insensitive logic (to_lower) in one arm.